### ISSUE TITLE
ApplicationWindow's set_startup_id causing SIGSEVs on Windows

### ISSUE THREAD

**Issue by Cogitri** [gtk-rs/gtk#811]
Hello there,

it appears that setting startup_id on Windows either directly or via Glade causes segfaults on windows. See https://github.com/Cogitri/gxi/pull/212/commits/fa5b0231520dad0af4562c2cde6d70ca6475ffc2.

https://developer.gnome.org/gtk3/stable/GtkWindow.html#gtk-window-set-startup-id tells me that this setting isn't relevant on anything but X11 (so anything but *NIX excluding MacOS), so maybe this should just be ignore entirely on Windows?

Sadly I can't provide a nice log for this since my Windows knowledge is basically 0, but I'd be glad to provide some of someone could tell me how :)

**Comment by EPashkin** [comment]
I confirm that `window.set_startup_id("sid1");` cause AV somewhere in GTK
```
error: process didn't exit successfully: `D:\eap\rust\0\./target\debug\basic.exe` (exit code: 0xc0000005, STATUS_ACCESS_VIOLATION)
```
`gdk::WindowExt::set_startup_id(&window.get_window().unwrap(), "sid1");` on visible window cause same error.

IMHO problem is that [gdk_window_set_startup_id](https://gitlab.gnome.org/GNOME/gtk/blob/gtk-3-24/gdk/gdkwindow.c#L10499) called implementation without check and no implementation for it in v3.24
 https://gitlab.gnome.org/GNOME/gtk/blob/gtk-3-24/gdk/win32/gdkwindow-win32.c#L6239

We can add `cfg_condition` to these functions but it don't helps with glade at all

**Comment by sdroege** [comment]
Yeah this is a bug in GTK. Preparing a MR for that right now but of course that's not going to help until you update to a future release.

I don't think there's anything we could do about this here.

**Comment by Cogitri** [comment]
Alright, wasn't sure about that. Will report it upstream then, thanks! :)

**Comment by sdroege** [comment]
No need to report upstream, doing so already :)

**Comment by Cogitri** [comment]
Thanks! :)

**Comment by sdroege** [comment]
See https://gitlab.gnome.org/GNOME/gtk/merge_requests/825

**Comment by sdroege** [comment]
This is fixed in GTK 3.24.9 and above.
